Release/2025.08

Geenz opened the pull request:
as of = 2025-11-18 for RC deploy
build = https://github.com/secondlife/viewer/releases/tag/Second_Life_Release%23ad6008a5-2025.08
cohort = 2025.08
deployed = https://github.com/secondlife/viewer/releases/tag/Second_Life_Release%23ad6008a5-2025.08
desired = 3000
relnotes:

### Second Life Viewer 2025.08 Crash & Stability Fixes

Hey everyone\! We’re excited to share Second Life Viewer 2025.08! This is a smaller release that focuses on reliability, crash fixes, and small quality-of-life improvements.

#### Quality of Life Improvements

* **AM/PM Time Selector:** Added a new selector to improve time readability in the UI.
* **Improved Roles & Members Tab:** Layout adjustments make group management a bit clearer and easier to navigate.
* **Chat Popups:** Clicking a chat notification now opens the torn-off chat window directly.
* **Graphics Control:** You can now change graphics quality via a debug setting.
* **Mouse Warp Option:** The Viewer UI now exposes mouse warp controls for advanced camera workflows.
* **Default Material Handling:** Fixed logic when clearing materials — you can now properly set a material to *Default* or *None*.
* **Fixes for My Outfits:** Improved performance when loading or switching outfits.
* **Mic Volume Slider:** Preferences mic volume control now correctly adjusts your voice input.
* **Japanese Translation:** Updated and corrected several strings. (Thanks @sakuranoelfayray\!)

#### Technical Fixes & Performance

* **Crash & Freeze Handling:** Improved internal detection to better distinguish freezes from crashes.
* **AMD Radeon 8060S Freeze:** Added a work around for an AMD driver bug that freezes the viewer on startup.  Note: performance may be impacted temporarily.
* **WebRTC Stability:** Multiple fixes to prevent voice and video-related crashes during shutdown.
* **GLTF Uploads:** Fixed upload issues when filenames contained spaces.
* **OpenJPEG Textures:** Corrected small texture upload errors.
* **My Outfits:** Addressed performance issues caused by excessive fetching and sorting.
* **Texture & Mesh Handling:**
  * Fixed baked texture sharing between avatars.
  * Improved sub-mesh ordering on upload.
  * Preserved transforms when switching PBR materials.
  * Resolved inconsistencies in GLTF and Collada naming conventions.
* **Cache & Build Improvements:**
  * Removed old `cef_cache` folders when moving cache directories.
  * Consistent `LLFile` behavior across Windows, Linux, and macOS.
* **System Stability:**
  * Fixed crashes during headset unplug/replug events on macOS.
  * Corrected issues with `LLVolumeFace`, `LLPluginProcessParent`, and `LLCoprocedureManager` shutdowns.
  * Fixed window session termination handling.
* **Audio Device UI:** Fixed issue where device list appeared filled with zeros.
* **Texture Logging:** Reduced excessive logging during shutdown and improved crash diagnostics.
* **Login & Display Name:**
  * Extended login timeout for slow connections.
  * Fixed cases where users couldn’t reset their display name.

#### Developer & Build Updates

* NSIS, pre-commit, and universal build fixes across macOS and Windows.
* Consistent file handling between operating systems.
* Updated `actions/checkout` workflow to v5.
* Added new VHACD-based convex decomposition library.
* Fixed Havok integration in macOS builds.
* Pre-commit checks for whitespace and tab consistency.

#### Contributors

**New contributors this release:**

* [@Aqil-Ahmad](https://github.com/Aqil-Ahmad) - PBR Material Transform Preservation
* [@sakuranoelfayray](https://github.com/sakuranoelfayray) - Japanese Translation Fixes

Big thanks to all contributors and QA for keeping this release stable and performant!

LLGuru committed:
#2618 Roles and Members tab is a mess (fix XML indents for readability)

LLGuru committed:
#2618 Roles and Members tab is a mess

akleshchev committed:
viewer#2172 AM/PM selector

LLGuru committed:
#2577 Worn transparent prims no longer show with 'Highlight transparent'

cosmic-linden committed:
secondlife/viewer#2570: Fix baked texture shared between avatars when in PBR material

akleshchev committed:
viewer#3018 Expose mouse wrap in UI

And attempt some basic automation

akleshchev committed:
#1965 Properly terminate particles when HUD is off #2

akleshchev committed:
#1825 Use html as a default type for nosniff mime probes

akleshchev committed:
#4651 Handle window's sessions termination

akleshchev committed:
#4660 Unable to upload GLTF+bin with spaces in the name

akleshchev committed:
#4651 Try distinguishing freezes from crashes

marchcat committed:
LLLeap: handle partial lines in stderr (#4678)

akleshchev committed:
#789 Cannot set the material to Default or None

After setting material to Blank when blank material is present in inventory

akleshchev committed:
#4651 Fix marker presence check

maxim-productengine committed:
#4432 Change graphics quality via a debug setting

akleshchev committed:
#4685 EventQueueGet shouldn't use llcorehttp's backoff-retry logic

- Event poll already has own backoff logic with 10 retries, using llcorehttp's one on top makes no sense.
- Better logging coverage
- Better in-code documentation
- Add timing-based error detection

akleshchev committed:
#4687 Fix audio device selection showing a bunch of zeros

RyeMutt committed:
Initial VHACD based llconvexdecomposition

akleshchev committed:
#4671 Some locales are missing AM/PM labels

marchcat committed:
NSIS build fix (#4698)

Geenz committed:
Merge remote-tracking branch 'origin/release/2025.07' into geenz/2025.07-develop

Geenz committed:
Merge pull request #4700 from secondlife/geenz/2025.07-develop

Geenz/2025.07 -> develop

akleshchev committed:
#4679 Logging for crash on updateMenuOptions

Logs say that viewer closed settings picker and immediately crashed
Yet callstack indicates that some inventory was right clicked, which
shouldn't be possible if picker already closed.
May be some click is closing the picker and opens menu at the same time,
but it's better to gather information first.

Hecklezz committed:
Fix OpenJPEG uploading small textures by correctly accounting for the maximum numresolutions

akleshchev committed:
Fix builds after update of build-variables branch

maxim-productengine committed:
#4715 disable ZoomIn an object item when the region is not connected to the current region

akleshchev committed:
#4724 Fix performance problems with My Outfits

akleshchev committed:
SL-20598 Fix user not being able to reset display name

akleshchev committed:
#4730 Extend login timeout

maxim-productengine committed:
#4733 When moving the cache, the old cef_cache should be removed

akleshchev committed:
#4515 Improve ordering of sub-meshes upon upload

akleshchev committed:
#4723 Fix emoji warnings and missing emojis

maxim-productengine committed:
#p473 fix XUI warnings

akleshchev committed:
p#475 Excessive texture fetch logging on shutdown

akleshchev committed:
#4739 Wearables sort order reliability

Commit:
Merge branch 'release/2025.07' into develop

akleshchev committed:
#4750 Crash in LLToolBarView::handleDropTool

akleshchev committed:
#4753 Crash at LLScriptFloater::setNotificationId

akleshchev committed:
#3189 Fix missed else condition

akleshchev committed:
#4762 Better cacheOptimize's issue handling

akleshchev committed:
#4758 Extra logging to track discard 6

Plus clamped some values and preventing decode from starting

maxim-productengine committed:
#vp484 Crash in LLFolderViewFolder::setOpen

akleshchev committed:
#4768 Crash at retransmitUnackedPackets

For some reason there are no logs for these crashes, but lack of
connection data indicates that viewer was disconnected prior to crash.

akleshchev committed:
#4770 Crashes on ~LLCoprocedureManager()

Situation is not clear, so hoping that explicit
cleanup will reveal something.

akleshchev committed:
#4773 Crash on sub_image_lines

Just caught it and discard_level is somehow 7, which seems like it resulted in src going out of bounds, which crashed glTexSubImage2D

RyeMutt committed:
Fix infinite growth of mLoadingPhysicsShapes in mesh repository

akleshchev committed:
#4777 Fix webrtc crashing in worker threads on shutdown

Commit:
Add workflow_dispatch trigger to build workflow (#4774)

Allow builds to be manually triggered

akleshchev committed:
#1759 Fix texture picker using IMG_WHITE for blank materials

Aqil-Ahmad committed:
Fix #4195: Preserve transforms when switching PBR materials (#4725)

Fixes texture transforms being reset when switching from Blinn-Phong to PBR materials and between PBR. Previously, custom scale, offset, and rotation settings would be lost, making it tedious to switch between PBR materials.

akleshchev committed:
#4195 Clear code from a merge conflict

This part was removed unintentionally, due to a merge conflict resolution.

akleshchev committed:
p#480 Collada vs GLTF Mesh Import Naming Convention

Both collada and gltf have a node and a mesh. Collada uses node-name, gltf was using mesh-name.
GLTF format permits reusing single mesh for multiple nodes, but nodes are warrantied to not be reused.

Switch to using node-names for better dupplicate avoidance and to be more in line with collada.

akleshchev committed:
#4794 SceneLoadRearMaxRadiusFraction not stored as a fraction

maxim-productengine committed:
#4772 fix group ownership detection for deeded objects

akleshchev committed:
#4796 Preferences' mic volume slider had no effect.

akleshchev committed:
#4782 Improve wearing multiple items

akleshchev committed:
p#351 Mark RenderMaxOpenGLVersion as Windows specific

akleshchev committed:
#4803 Fix crash on cameraUnderWater

updateEnvironment tried to access dead camera

akleshchev committed:
#4806 Crash at getPrimaryFormat

According to bugsplat tex is null

akleshchev committed:
#4777 Fix crash on missing factory

Geenz committed:
Merge pull request #4809 from secondlife/andreyk/viewer_4777

#4777 Fix crash on missing factory

Geenz committed:
Merge pull request #4783 from RyeMutt/rye/convexdecomp

Introduce initial VHACD based llconvexdecomposition library

Hecklezz committed:
#4811 #4813 Fixed RemoveItemWarn invalid notification and deleting no-mod content (#4812)

* Fixed trying to display an invalid notification for RemoveItemWarn
* Fixed allowing deleting no-mod content if tried to be deleted twice

akleshchev committed:
#2818 Creating a link to an empty notecard fails

sakuranoelfayray committed:
Fix Japanese Translation (#743)

marchcat committed:
Avoid stack-buffer overruns in llprocess_test.cpp (#4824)

Ansariel committed:
Fix erroneous check for __attribute__ definition introduced with APR 1.7.5

With the update of APR to versoin 1.7.5., we have to deal with an erroneous check for the definition of __attribute__ added to the APR head file apr.h: The additional check for the definition of __has_attribute is nonsense as it is a C++ internal macro to check for the existance of certain attributes, however it has nothing to do with __attribute__ used in this context.

Since __attribute__ is not defined in MSVC, the modified, erroneous check now causes said definition becoming missing, resulting in a huge amount of "this declaration may not have extern 'C' linkage" errors within in MSVC.

This change utilizes CMake to patch the extracted APR header file, restoring the original version of the check.

akleshchev committed:
#3971 Fix wrong material type when 'editing linked'

akleshchev committed:
#3263 Repeat Notification Noise on Find Emoji Floater

RolfKal committed:
Improve LLFile to be consistent between Windows and Linux/Mac

Ansariel committed:
Fix C26444 compiler warning (creating unnamed RAII instances that could lead to unintended behavior)

akleshchev committed:
p#490 Fix missing Lossless check

that broke sculpt upload

RolfKal committed:
Remove a trailing space and a missed LLFile::remove() call right before the LLFile::rename()

akleshchev committed:
#2570 Fix extra setGLTFRenderMaterial

setRenderMaterialID already calls setGLTFRenderMaterial

akleshchev committed:
#4757 Fix missing quotes

akleshchev committed:
#3379 Fix callback cleanup

Instead of marking as 0, mark for cleanup

fmartian committed:
#2531 Fix gray background tiles in the Minimap (#4815)

akleshchev committed:
#4701 Fix incorrect architecture in package's name

RolfKal committed:
Remove trailing space

Geenz committed:
Merge pull request #4842 from secondlife/andreyk/viewer_4701

#4701 Fix incorrect architecture in package's name

Geenz committed:
Merge remote-tracking branch 'origin/main' into geenz/main-to-develop

Geenz committed:
Merge pull request #4850 from secondlife/geenz/main-to-develop

Catch some commits that didn't make it back to develop.

akleshchev committed:
Bump viewer version tp 7.2.3 after 2025.07 release

fmartian committed:
Make changes according to recommendations by Copilot
- correction spelling of suppress_error
- improved error handling in remove() based on functionality in get_fileattr() and somewhat changed error handling in get_fileattr() itself
- call explicitly LLFile::fopen() to make sure we use the correct file path conversion under Windows

Removing Flawfinder comments since Flawfinder isn't used in the viewer anymore
Adding an option to support symlink detection in getattr()
Adding comments to function implementation to indicate that they are really static functions of the LLFile class

RyeMutt committed:
Ignore tabs to spaces for objc in blame

RyeMutt committed:
Update pre-commit hooks to v6.0.0

RyeMutt committed:
Fix tabs and trailing spaces in objc sources

RyeMutt committed:
Introduce dependabot configuration for Github Actions (#4859)

akleshchev committed:
#4861 Crash at LLVolumeFace::createOctree

Looks like LLRiggedVolume::update goes out of bounds when selecting a face

RyeMutt committed:
Rework new convex decomp into a physicsextensions stub package and fix havok and havok_tpv builds for darwin universal to utilize new vhacd stub (#4858)

fmartian committed:
Fix typo in comment

dependabot[bot] committed:
Bump pre-commit/action from 3.0.0 to 3.0.1 (#4864)

Bumps [pre-commit/action](https://github.com/pre-commit/action) from 3.0.0 to 3.0.1.
- [Release notes](https://github.com/pre-commit/action/releases)
- [Commits](https://github.com/pre-commit/action/compare/v3.0.0...v3.0.1)

---
updated-dependencies:
- dependency-name: pre-commit/action
  dependency-version: 3.0.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/checkout from 4 to 5 (#4867)

Bumps [actions/checkout](https://github.com/actions/checkout) from 4 to 5.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

akleshchev committed:
#2427 Show torn off chat when clicking on a message popup

akleshchev committed:
#4871 Fix webrtc crash on tuningGetEnergy

akleshchev committed:
#4839 Crash at LLPluginProcessParent::poll

Missed lock

akleshchev committed:
Merge from secondlife/develop into 2025.08

akleshchev committed:
#4876 Fix material not applying

1. Dupplicate call in llselectmgr can cause race conditions
2. lltooldragndrop wasn't updating server in one case

akleshchev committed:
#4884 Crash at resetRegionCrossingTimer

akleshchev committed:
#4886 Crash on LLTeleportRequestViaLure::toOstream

akleshchev committed:
#4879 An Emoji in a display name can affect text in chat

An emoji was spliting inlineview thus affecting displayed text.

roxanneskelly committed:
Fix issue where mac was crashing during an attempt to unplug or replug microphone. (#4897)

The mac audio device manager was being "helpful" by restarting
playout and recording if the Default device was changed, assuming
the application wouldn't care.
However, we received an update of device change, and attempted to
stop and start playout anyway, causing a conflict.
The fix was simply to not deploy new devices when the device id didn't
change.

akleshchev committed:
#4882 Log textures that failed to be created

A bunch of 36x36 failed to create, there shouldn't have been any 36x36
textures, log the ids/type.

akleshchev committed:
#4604 Turn off cef preloading on low end hardware

1. Cef instances are fairly heavy in viewer's implementation, don't preload
them on systems <= 8GB
2. Move initialization into floaters where it belongs.
3. Minimal requirement is 8GB, drop gpu class to 1 on hardware under 8GB
4. Reduce cef instance count on systems under 8GB

Dan-Linden commented:
| 2025.08 Beta 7.2.3.18887079901 | Result | Comment |
|---|---|---|
| secondlife/iqa#563 Acceptance, Integration & Additional Tests | Passed | Passed |
| **Overall Result** | **Passed** | **OK to ship** |

**NOTES**
Simple smoke and Acceptance testing passed.
Most of the bug fixes have not yet been verified.

Note about the AV: Reprocessing slplugin.exe on virustotal sometimes triggers Microsoft - Trojan:Win32/Wacatac.B!ml and sometimes triggers Program:Win32/Wacapew.C!ml. The "!ml" suffix findings in Microsoft Defender are Machine Learning (AI or heuristic) findings.

Anti-virus false positive report:
| File | Security vendor analysis |
|---|---|
| SLVersionChecker.exe | SecureAge - Malicious |
| slplugin.exe | MaxSecure - Trojan.Malware.300983.susgen<br> Microsoft - Trojan:Win32/Wacatac.B!ml (sometimes Program:Win32/Wacapew.C!ml)<br> SecureAge - Malicious |

kylelinden reviewed (approved):
2025.08 Beta Crash Fixes approved.
viewer = 7.2.3.18887079901
build = https://github.com/secondlife/viewer/releases/tag/Second_Life_Release%235a0ba25d-2025.08
cohort = 2025.08
desired = 1000

Geenz commented:
deployed = https://github.com/secondlife/viewer/releases/tag/Second_Life_Release%235a0ba25d-2025.08

maxim-productengine committed:
#4847 log session_jitter and period_jitter with FPS info

Ansariel committed:
Fix XUI parser warnings:

* LLFloaterMarketplace::postBuild calls parent method which already tries to find the webbrowser control with a different name and adds itself as observer
* Incorrect attribute name "tooltip" in panel_preferences_move.xml

kylelinden commented:
Reduced cohort to 500 due to early session freeze https://github.com/secondlife/viewer/issues/4918

akleshchev committed:
#4918 Crash at LLPluginProcessParent::poll

Bugsplat shows that mutex is null.
It's possible that no instances exist thus mutex wasn't initialized.

Dan-Linden commented:
| 2025.08 Beta 7.2.3.18980112811 | Result | Comment |
|---|---|---|
| secondlife/iqa#565 Acceptance, Integration & Additional Tests | Passed | Passed |
| **Overall Result** | **Passed** | **OK to ship** |

**NOTES**
This build has a fix for blocker https://github.com/secondlife/viewer/issues/4918.
Simple smoke and Acceptance testing passed.
Most of the bug fixes have not yet been verified.

Anti-virus false positive report:
| File | Security vendor analysis |
|---|---|
| SLVersionChecker.exe | SecureAge - Malicious |
| slplugin.exe | MaxSecure - Trojan.Malware.300983.susgen<br> SecureAge - Malicious |

maxim-productengine committed:
#4920 fix group categories not being displayed when one below is expanded

kylelinden reviewed (approved):
2025.08 Beta crash fixes approved
viewer = 7.2.3.18980112811
build = https://github.com/secondlife/viewer/releases/tag/Second_Life_Release%23f7516a46-2025.08
cohort = 2025.08
desired = 1000

Ansariel committed:
Use correct widget name for marketplace floater

Ansariel committed:
Fix remaining cases of webbrowser-related XUI warnings

Geenz commented:
deployed = https://github.com/secondlife/viewer/releases/tag/Second_Life_Release%23f7516a46-2025.08

akleshchev committed:
#4785 The Apply button is not activated after changing a Material

akleshchev committed:
#1006 Assume default material is valid for terrain assigment

akleshchev committed:
#4656 Don't scroll to a child that is already in view and can not fit

akleshchev committed:
#4651 Distinguish logout crashes from other crashes

Dan-Linden commented:
| 2025.08 Beta 7.2.3.19118091373 | Result | Comment |
|---|---|---|
| secondlife/iqa#566 Acceptance, Integration & Additional Tests | Passed | Passed |
| **Overall Result** | **Passed** | **OK to ship** |

**NOTES**
Simple smoke and Acceptance testing passed.
Some of the bug fixes have not yet been verified.

Anti-virus false positive report:
| File | Security vendor analysis |
|---|---|
| SLVersionChecker.exe | SecureAge - Malicious |
| slplugin.exe | MaxSecure - Trojan.Malware.300983.susgen<br> SecureAge - Malicious |

kylelinden reviewed (approved):
2025.08 Beta Crash fixes
viewer = 7.2.3.19118091373
build = https://github.com/secondlife/viewer/releases/tag/Second_Life_Release%237a7215b8-2025.08
cohort = 2025.08
desired = 2000

Geenz commented:
deployed = https://github.com/secondlife/viewer/releases/tag/Second_Life_Release%237a7215b8-2025.08

akleshchev committed:
#4949 QueuedRequest destructor crash

Geenz committed:
Disable shader profiles on RDNA3.5 to prevent freezes on startup. (#4992)

* Disable shader profiling for certain AMD Radeon GPUs

Added logic to detect AMD Radeon 8060 GPUs and disable shader profiling to prevent client freezes and instability. Introduced sCanProfile flag in LLGLSLShader and mSkipProfiling in LLFeatureManager to control profiling behavior based on detected hardware.

* Add RDNA3.5 and check the vendor string for a known current family of bad drivers

* Update llfeaturemanager.cpp

* Make sure to check that this is a Radeon.

Geenz committed:
Disable the benchmark & occlusion on RDNA3.5 (#4999)

Dan-Linden commented:
| 2025.08 Beta 7.2.3.19375695301 | Result | Comment |
|---|---|---|
| secondlife/iqa#568 Acceptance, Integration & Additional Tests | Passed | Passed |
| **Overall Result** | **Passed** | **OK to ship** |

**NOTES**
Simple smoke and Acceptance testing passed.
2 of the bug fixes have not yet been verified.

Anti-virus false positive report:
| File | Security vendor analysis |
|---|---|
| SLVersionChecker.exe | SecureAge - Malicious |
| slplugin.exe | MaxSecure - Trojan.Malware.300983.susgen<br> SecureAge - Malicious |

kylelinden reviewed (approved):
2025.08 Beta crash fixes approved
viewer = 7.2.3.19375695301
build = https://github.com/secondlife/viewer/releases/tag/Second_Life_Release%23ad6008a5-2025.08
cohort = 2025.08
desired = 3000

Geenz commented:
deployed = https://github.com/secondlife/viewer/releases/tag/Second_Life_Release%23ad6008a5-2025.08

kylelinden commented:
Approved for promotion to default release.
